File: airline/flutter_app/lib/models/aircraft.dart
class Aircraft {
  final int id;
  final String model;
  final String registrationNumber;
  final int capacity;
  final int seatTemplateId;

  Aircraft({
    required this.id,
    required this.model,
    required this.registrationNumber,
    required this.capacity,
    required this.seatTemplateId,
  });

  factory Aircraft.fromJson(Map<String, dynamic> json) {
    return Aircraft(
      id: json['id'],
      model: json['model'],
      registrationNumber: json['registration_number'],
      capacity: json['capacity'],
      seatTemplateId: json['seat_template_id'] ?? 
                      (json['seat_template'] != null ? json['seat_template']['id'] : 1),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'model': model,
      'registration_number': registrationNumber,
      'capacity': capacity,
    };
  }

  String get displayName => '$model ($registrationNumber)';
}

File: airline/backend/app/models/aircraft.py
from sqlalchemy import Column, Integer, String, ForeignKey, JSON
from sqlalchemy.orm import relationship
from app.core.database import Base


class Aircraft(Base):
    __tablename__ = "aircrafts"

    id = Column(Integer, primary_key=True, index=True)
    model = Column(String, nullable=False)
    registration_number = Column(String, unique=True, nullable=False)
    capacity = Column(Integer, nullable=False)
    seat_template_id = Column(Integer, ForeignKey("seat_templates.id"), nullable=False)

    # Relationships
    seat_template = relationship("SeatTemplate", back_populates="aircrafts")
    flights = relationship("Flight", back_populates="aircraft")


class SeatTemplate(Base):
    __tablename__ = "seat_templates"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    
    # Configuration for generator
    row_count = Column(Integer, nullable=False, default=20)
    seat_letters = Column(String, nullable=False, default="ABC DEF") # Letters with space for aisle
    business_rows = Column(String, nullable=True) # e.g. "1-3"
    economy_rows = Column(String, nullable=True)  # e.g. "4-20"
    
    seat_map = Column(JSON, nullable=False)  # Generated JSON structure

    # Relationships
    aircrafts = relationship("Aircraft", back_populates="seat_template")



